pkg/skaffold/schema/profile_test.go

```diff
@@ -142,13 +156,164 @@ func TestApplyProfiles(t *testing.T) {
 				withHelmDeploy(),
 			),
 		},
+		{
+			description: "patch Dockerfile",
+			profile:     "profile",
+			config: config(
+				withLocalBuild(
+					withGitTagger(),
+					withDockerArtifact("image", ".", "Dockerfile"),
+				),
+				withKubectlDeploy("k8s/*.yaml"),
+				withProfiles(latest.Profile{
+					Name: "profile",
+					Patches: yamlpatch.Patch{{
+						Path:  "/build/artifacts/0/docker/dockerfile",
+						Value: yamlpatch.NewNode(str("Dockerfile.DEV")),
+					}},
+				}),
+			),
+			expected: config(
+				withLocalBuild(
+					withGitTagger(),
+					withDockerArtifact("image", ".", "Dockerfile.DEV"),
+				),
+				withKubectlDeploy("k8s/*.yaml"),
+			),
+		},
+		{
+			description: "invalid patch path",
+			profile:     "profile",
+			config: config(
+				withLocalBuild(
+					withGitTagger(),
+					withDockerArtifact("image", ".", "Dockerfile"),
+				),
+				withKubectlDeploy("k8s/*.yaml"),
+				withProfiles(latest.Profile{
+					Name: "profile",
+					Patches: yamlpatch.Patch{{
+						Path: "/unknown",
+						Op:   "replace",
+					}},
+				}),
+			),
+			shouldErr: true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.description, func(t *testing.T) {
+			err := ApplyProfiles(test.config, &cfg.SkaffoldOptions{
+				Profiles: []string{test.profile},
+			})
+
+			if test.shouldErr {
+				testutil.CheckError(t, test.shouldErr, err)
+			} else {
+				testutil.CheckErrorAndDeepEqual(t, test.shouldErr, err, test.expected, test.config)
+			}
+		})
+	}
+}
+
+func TestActivatedProfiles(t *testing.T) {
+	tests := []struct {
+		description string
+		profiles    []latest.Profile
+		opts        *cfg.SkaffoldOptions
+		expected    []string
+		shouldErr   bool
+	}{
+		{
```

Comment:
There are no tests covering the case where the `Activation` array has multiple activations. It is unclear if it should be OR or AND  - it looks like from the implementation that it's a logical OR.